pkg/controller/rolloutblock/rolloutblock_controller.go

```diff
@@ -0,0 +1,622 @@
+			workqueue.DefaultControllerRateLimiter(),
+			"rolloutblock_controller_rolloutblocks",
+		),
+	}
+
+	glog.Info("Setting up event handlers")
+
+	releaseInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.addRelease,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				controller.updateRelease(oldObj, newObj)
+			},
+			DeleteFunc: controller.deleteRelease,
+		})
+
+	applicationInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.addApplication,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				controller.updateApplication(oldObj, newObj)
+			},
+			DeleteFunc: controller.deleteApplication,
+		})
+
+	rolloutblockInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.addRolloutBlock,
+		})
+
+	return controller
+}
+
+func (c *Controller) addRelease(obj interface{}) {
+	rel, ok := obj.(*shipper.Release)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("not a shipper.Release: %#v", obj))
+		return
+	}
+
+	key, err := cache.MetaNamespaceKeyFunc(rel)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	overrideRB, ok := rel.GetAnnotations()[shipper.RolloutBlocksOverrideAnnotation]
+	if !ok {
+		overrideRB = ""
+		return
+	}
+
+	overrideRBs := strings.Split(overrideRB, ",")
+	for _, rbKey := range overrideRBs {
+		rolloutBlockUpdater := RolloutBlockUpdater{
+			rbKey,
+			OverridingRelease,
+			key,
+			false,
+		}
+		c.rolloutblockWorkqueue.Add(rolloutBlockUpdater)
+	}
+}
+
+func (c *Controller) deleteRelease(obj interface{}) {
+	release, ok := obj.(*shipper.Release)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("not a shipper.Application: %#v", obj))
+		return
+	}
+
+	key, err := cache.MetaNamespaceKeyFunc(release)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	overrideRB, ok := release.GetAnnotations()[shipper.RolloutBlocksOverrideAnnotation]
+	if !ok {
+		overrideRB = ""
+		return
+	}
+
+	overrideRBs := strings.Split(overrideRB, ",")
+	for _, rbKey := range overrideRBs {
+		rolloutBlockUpdater := RolloutBlockUpdater{
+			rbKey,
+			OverridingRelease,
+			key,
+			true,
+		}
+		c.rolloutblockWorkqueue.Add(rolloutBlockUpdater)
+	}
+}
+
+func (c *Controller) updateRelease(oldObj, newObj interface{}) {
+	oldRel, ok := oldObj.(*shipper.Release)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("not a shipper.Release: %#v", oldObj))
+		return
+	}
+
+	newRel, ok := newObj.(*shipper.Release)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("not a shipper.Release: %#v", newObj))
+		return
+	}
+
+	newOverride, newOk := newRel.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+	oldOverride, oldOk := oldRel.Annotations[shipper.RolloutBlocksOverrideAnnotation]
+	if !oldOk {
+		oldOverride = ""
+	}
+	if !newOk {
+		newOverride = ""
+	}
+
+	newOverrides := strings.Split(newOverride, ",")
+	oldOverrides := strings.Split(oldOverride, ",")
+	if stringUtil.Equal(newOverrides, oldOverrides) {
```

Comment:
if we compose them in a deterministic way (which is the case if we stable-sort them), a simple string comparison is enough without parsing and a granular per-element comparison